src/object/srv_obj.c

```diff
@@ -1539,8 +1539,7 @@ obj_local_rw_internal(crt_rpc_t *rpc, struct obj_io_context *ioc,
 		if (ioc->ioc_coc->sc_props.dcp_csum_enabled) {
 			rc = csum_add2iods(ioh,
 					   orw->orw_iod_array.oia_iods,
-					   orw->orw_iod_array.oia_iod_nr,
-					   ioc->ioc_coc->sc_csummer,
+					   iods_nr, ioc->ioc_coc->sc_csummer,
```

Comment:
why "orw->orw_iod_array.oia_iods" instead of the input "iods"? that is the pairs of "iods_nr".

Reply:
So we may add "D_ASSERT(orw->orw_iod_array.ioa_iods should be always == iods)" ? if has other chance to refresh the patch?

Anyway, your patches seems not resolve the issue:

https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-11624/2/artifact/Functional%20Hardware%20Medium/scrubber/target_auto_eviction.py/stacktraces.wolf-138/core.daos_io_1.1678316154.114222.stacktrace/*view*/

Thread 1 (Thread 0x7f176168f700 (LWP 114471)):
#0  0x00007f17dfe5fa9f in raise () from /lib64/libc.so.6
No symbol table info available.
#1  0x00007f17dfe32e05 in abort () from /lib64/libc.so.6
No symbol table info available.
#2  0x00007f17dfe32cd9 in __assert_fail_base.cold.0 () from /lib64/libc.so.6
No symbol table info available.
#3  0x00007f17dfe583f6 in __assert_fail () from /lib64/libc.so.6
No symbol table info available.
#4  0x0000559e25a607cd in proc_struct_dcs_csum_info_adv (proc=proc@entry=0x7f175ea29f10, proc_op=proc_op@entry=CRT_PROC_ENCODE, csum=0x7f175ed16810, idx=idx@entry=0, nr=<optimized out>) at src/object/rpc_csum.c:65
        buf_len = 2
        rc = 0
        __func__ = "proc_struct_dcs_csum_info_adv"
        __PRETTY_FUNCTION__ = "proc_struct_dcs_csum_info_adv"
#5  0x0000559e25a60d4b in proc_struct_dcs_csum_info (proc=proc@entry=0x7f175ea29f10, proc_op=proc_op@entry=CRT_PROC_ENCODE, csum=<optimized out>) at src/object/rpc_csum.c:95
No locals.
#6  0x0000559e25a60de3 in proc_struct_dcs_csum_info (csum=<optimized out>, proc_op=CRT_PROC_ENCODE, proc=0x7f175ea29f10) at src/object/rpc_csum.c:92
No locals.
#7  crt_proc_struct_dcs_csum_info (proc=proc@entry=0x7f175ea29f10, proc_op=<optimized out>, p_csum=p_csum@entry=0x7f175ea8b958) at src/object/rpc_csum.c:110
        csum_enabled = true
        rc = 0
        __func__ = "crt_proc_struct_dcs_csum_info"
#8  0x0000559e25a5190e in crt_proc_obj_rw_in (proc=0x7f175ea29f10, ptr=0x7f175ea8b8b0) at src/object/obj_rpc.c:1079
        proc_op = CRT_PROC_ENCODE
        rc = 0
        __func__ = "crt_proc_obj_rw_in"

Reply:
This csum assertion is different issue DAOS-12857.